collectors/python.d.plugin/nvidia_smi/nvidia_smi.chart.py

```diff
@@ -326,6 +336,20 @@ def name(self):
     def full_name(self):
         return 'gpu{0} {1}'.format(self.num, self.name())
 
+    @handle_attr_error
+    def pci_link_gen(self):
+        return self.root.find('pci').find('pci_gpu_link_info').find('pcie_gen').find('max_link_gen').text
+
+    @handle_attr_error
+    def pci_link_width(self):
+        return self.root.find('pci').find('pci_gpu_link_info').find('link_widths').find('max_link_width').text.split('x')[0]
+
+    def pci_bw_max(self):
+        link_gen = int(self.pci_link_gen())
+        link_width = int(self.pci_link_width())
+        # return max bandwidth in kB/s
+        return link_gen * 250 * 1024 * link_width
```

Comment:
Can you explain (or share a link to the doc) this calculation? What is 250?

I [googled](https://enterprise-support.nvidia.com/s/article/understanding-pcie-configuration-for-maximum-performance#:~:text=Maximum%20PCIe%20Bandwidth%20%3D%20SPEED%20*%20WIDTH,1G%20%3D%20~62Gb%2Fs.):

> The maximum possible PCIe bandwidth is calculated by multiplying the PCIe width and speed. From that number, we reduce ~1Gb/s for error correction protocols and the PCIe headers overhead. The overhead is determined by both the PCIe encoding (see PCIe speed for details), and the PCIe MTU:

> **Maximum PCIe Bandwidth = SPEED * WIDTH * (1 - ENCODING) - 1Gb/s**.

Reply:
Indeed the calculation is not right, I need to rework it.